### PR TITLE
Fixed WPML terms ids not matching with term taxonomy ids

### PR DESCRIPTION
**Pull request by marcdap**
### Original source:
- https://github.com/netzstrategen/haur-digi

### Task:
- [Bug: Menues](https://app.asana.com/0/1141864033797921/1203249137170234)

### WPML Support links:
- https://wpml.org/forums/topic/the-position-of-translate-menu-is-not-working/#post-13399395
- https://wpml.org/forums/topic/unable-to-set-menu-location-with-wpml-active/

### Description
For some languages (Italian and Swiss) seems that the taxonomy terms' ids were not matching with the current terms' ids. This was causing an issue that for these languages the position was not correctly set, creating some relationship between different items from different menus. This was causing moving an item from the header was moving also items from the footer.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204183116014945

### CODE REVIEW

File: patches/search-exclude.0001.fix.posts-exclude-overwritten.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+         if ($exclude) {
+-            $query->set('post__not_in', array_merge(array(), $this->getExcluded()));
++            $excluded_ids = $query->query_vars['post__not_in'] ?? [];
++            $query->set('post__not_in', array_unique(array_merge($excluded_ids, $this->getExcluded())));
+         }
+ 
+         return $query;
+-- 
+2.34.1
+
```
Comment (navidabdi) [inline patches/search-exclude.0001.fix.posts-exclude-overwritten.patch:26]: Is this patch related to the menu fix?

**Review by navidabdi — CHANGES_REQUESTED** [review]

File: patches/search-exclude.0001.fix.posts-exclude-overwritten.patch (hunk 1 above)
Comment (marcdap) [reply patches/search-exclude.0001.fix.posts-exclude-overwritten.patch:26]: @navidabdi it is not related to the patch. The problem is that Hauraton is pointing to this PR that has not been merged into main. I will update Hauraton to main, and rebase this branch on top. But it will have commits that are not related to this particular patch, but I think we cannot lose them.
